q2_moshpit/eggnog/_utils.py

```diff
@@ -26,3 +39,112 @@ def _parse_build_diamond_db_params(arg_key, arg_val) -> List[str]:
+                for i, file in tqdm(
+                    enumerate(files, start=1), total=len(files)
+                ):
+                    if file.endswith(".hmm"):
+
+                        # process hmm files
+                        with open(f"{root}/{file}", "r") as hmm_file:
+                            lines = hmm_file.readlines()
+
+                            # Find "NAME" line
+                            for j, line in enumerate(lines):
+                                if line.startswith("NAME "):
+                                    modified_line = re.sub(
+                                        r"\.faa\.final_tree(\.fa)?", "", line
+                                    )
+
+                                    # write modified content to hmms_merged
+                                    lines[j] = modified_line
+                                    hmms.writelines(lines)
+
+                                    # get name and write to idmap
+                                    id = modified_line.replace("NAME  ", "", 1)
+                                    idmap.write(f"{i} {id}")
+
+                                    break
+
+    # prepare an HMM database for faster hmmscan searches
+    print(colorify("Preparing HMM database..."))
+    run_command(cmd=["hmmpress", hmms_merged_p])
+    shutil.move(hmms_merged_p, f"{str(hmm_db_obj)}/{taxon_id}.hmm")
+
+    return idmap_obj, hmm_db_obj, pressed_hmm_db_obj
+
+
+def _download_fastas_into_hmmer_db(taxon_id: int):
+    fastas_obj = ProteinsDirectoryFormat()
+    with tempfile.TemporaryDirectory() as tmp:
+        _try_wget(
+            f"{tmp}/{taxon_id}_raw_algs.tar",
+            "http://eggnog5.embl.de/download/eggnog_5.0/per_tax_level/"
+            f"{taxon_id}/{taxon_id}_raw_algs.tar",
+            "Error downloading FASTA files"
+        )
+
+        # Extracting
+        print(colorify("Decompressing..."))
+        run_command(
+            cmd=["tar", "xf", f"{taxon_id}_raw_algs.tar"],
+            cwd=tmp
+        )
+
+        files = [
+            f"{tmp}/{taxon_id}/{f}"
+            for f in os.listdir(f"{tmp}/{taxon_id}")
+            if f.endswith(".gz")
+        ]
+
+        # Extract, remove '-' and save to hmmer_db location
+        print(colorify("Processing FASTA files (this can take a while)... "))
+        for fpi in tqdm(files):
+            new_name = os.path.basename(fpi).replace(".raw_alg.faa.gz", ".fa")
+            fpo = os.path.join(str(fastas_obj), new_name)
+            with gzip.open(fpi, "rt") as f_in, open(fpo, "w") as f_out:
+                content = f_in.read()
+                content = content.replace("-", "")
+                f_out.write(content)
+
+    return fastas_obj
+
+
+def _try_wget(output_file: str, url: str, exception_msg: str):
+    try:
+        run_command(cmd=["wget", "-O", output_file, url])
+    except subprocess.CalledProcessError as e:
+        raise Exception(
+            f"{exception_msg}: {e.returncode}"
+        )
```

Comment:
I think this function might save quite a few lines of code if we want to use it in the rest of the wget commands in the _dbs.py module. I think this would be good practice since until now we are not really supporting the case in wich the download fails. But perhaps its better to leave that to for MAINT PR.

Reply:
If you see my comment below on handling the failed download, it would give even more support to just placing this code directly where this function is being called. Yes, it will save you 3-4 lines of code, probably not such a big deal when you contrast it with the visibility and a more custom exception-handling. All the URLs you're using have a very long common prefix - you can just extract it into a constant and place it on top of the module - will save you some lines too ;)